Data/DataModel/Message.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace ServerCore.DataModel
+{
+    /// <summary>
+    /// A message.
+    /// </summary>
+    public class Message
+    {
+        /// <summary>
+        /// Gets or sets the id.
+        /// </summary>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int ID { get; set; }
+
+        /// <summary>
+        /// Gets or sets a thread id.
+        /// </summary>
+        public string ThreadId { get; set; }
```

Comment:
Do you expect that there will be multiple threads per (team-puzzle x team) or (single-player-puzzle x player) or (event x team) or (event x player)? If not, then you do not need this ThreadID. Personally I don't think it's necessary, but I won't quibble if you want it. If you do still want it, though, then I would suggest that you keep it an int, and have the ID be the ID of the start message (which, among other things, ensures that it's unique).

Reply:
I think this thread id will be good to have just in case we do need it.
It will also be useful for general non-puzzle related messages.

It makes sense from a "messaging" object.

Also, I plan to make the thread id for puzzle messages to be of the format "PuzzleMessage_<puzzleId>_<TeamId>"